jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py

```diff
@@ -269,13 +273,36 @@ def _on_flux_unit_changed(self, msg):
         else:
             self.flux_or_sb_selected = 'Surface Brightness'
 
+        # Always send a surface brightness unit to contours
+        if self.flux_or_sb_selected == 'Flux':
+            yunit = self._append_angle_correctly(yunit, self.angle_unit.selected)
+        self._find_and_convert_contour_units(yunit)
+
         # for displaying message that PIXAR_SR = 1 if it is not found in the FITS header
         if (
             len(self.app.data_collection) > 0
             and not self.app.data_collection[0].meta.get('PIXAR_SR')
         ):
             self.pixar_sr_exists = False
 
+    def _find_and_convert_contour_units(self, yunit=None):
+        if not yunit:
+            yunit = self.sb_unit_selected
+
+        if self.angle_unit_selected is None or self.angle_unit_selected == '':
+            # Can't do this before the plugin is initialized completely
+            return
+
+        for name, viewer in self._app._jdaviz_helper.viewers.items():
+            if not isinstance(viewer._obj, BqplotImageView):
+                continue
+            for layer in viewer._obj.state.layers:
+                # DQ layer doesn't play nicely with this attribute
+                if "DQ" in layer.layer.label:
+                    continue
```

Comment:
I'm guessing masks don't either.... could we instead check the units on the layer to make sure they're surface brightness?  Eventually we might need to generalize this method to also set display units for images in velocity, etc (from moment maps), but that isn't enabled in unit conversion yet, so is probably safe to ignore.

Reply:
I added a second check below for surface brightness - we know DQ doesn't work, so might as well do the fast check first for that still?